src/Components/Analyzers/src/DiagnosticDescriptors.cs

```diff
@@ -46,5 +46,14 @@ internal static class DiagnosticDescriptors
             DiagnosticSeverity.Warning,
             isEnabledByDefault: true,
             description: new LocalizableResourceString(nameof(Resources.ComponentParametersShouldBePublic_Description), Resources.ResourceManager, typeof(Resources)));
+
+        public static readonly DiagnosticDescriptor ComponentParametersShouldNotBeSetOutsideOfComponents = new DiagnosticDescriptor(
```

Comment:
I have to nitpick this too. It's about setting parameters of **other component types**.